### DISCUSSION TITLE
Where did delta_degC go, and will it ever return?

### DISCUSSION THREAD

**Discussion by MichaelTiemannOSC** [hgrecco/pint#1602] [Q&A] [answered]
I've been working with the development branch of `pint` to see whether more recent Measurement functionality can save me a lot of time working around np.isnan(measurement) issues.  I'm now running into this:

`pint.errors.UndefinedUnitError: 'delta_deg_Celsius' is not defined in the unit registry`

And there's a lot of commentary about the problems of temperature measurements, generally.  Any advice on how to walk on this thin ice while I can?  And whether I should plan to return to the shore on which I started, or hurry up to a better landing point?

**Comment by jules-ch** [comment, marked as answer]
Delta for `degree_Celsius` is still there.

```python
>>> from pint import UnitRegistry
>>> ureg = UnitRegistry()
>>> ureg.Quantity(1, "delta_degree_Celsius")
<Quantity(1, 'delta_degree_Celsius')>
```

`deg_Celsius` is not defined see the `default_en.txt` definition.

See line for aliases https://github.com/hgrecco/pint/blob/9db2b295e3bc4b5e67ce0f0b0651533281139844/pint/default_en.txt#L200

You can define a new alias if you want, see the documentation for that : https://pint.readthedocs.io/en/stable/defining.html

**Reply by MichaelTiemannOSC** [reply]
I found the errant text that introduced the wrong spelling of the unit.  Thanks!

**Comment by MichaelTiemannOSC** [comment]
Thanks, Jules!  I'm feeding off openscm-units and perhaps it's failing to get all the goodness out of the newly faceted pint world:

```                                                                                                                                                                                                                                                                                                                                                                                                                                                                                
"""                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   
This module handles initialization of pint functionality                                                                                                                                                                                                                                                                                                                                                                                                                                                                              
"""

import uncertainties
from pint import set_application_registry
from pint_pandas import PintArray, PintType
from openscm_units import unit_registry
import re

# openscm_units doesn't make it easy to set preprocessors.  This is one way to do it.                                                                                                                                                                                                                                                                                                                                                                                                                                                 
unit_registry.preprocessors=[
     lambda s1: re.sub(r'passenger.km', 'pkm', s1),
     lambda s2: s2.replace('BoE', 'boe'),
]

PintType.ureg = unit_registry
ureg = unit_registry
set_application_registry(ureg)
Q_ = ureg.Quantity
M_ = ureg.Measurement
PA_ = PintArray
```

**Comment by hgrecco** [comment]
Just as a minor comment, remember that delta units are not defined in the text file, but automatically created for all non-multiplicative units.

Regarding facets, my feeling is that now behavior is more compartmentalized and easier to hack. Once that #1595 gets reviewed and approved, the parser will be also stripped out and maybe the formatter. (with the new delegates addition).

**Comment by MichaelTiemannOSC** [comment]
Yes indeed, I noticed that in the source code.  I'm very excited to see the new version of Pint when it comes out!
